stellar-contract-env-host/src/test.rs

```diff
@@ -89,3 +89,31 @@ fn vec_as_seen_by_host() -> Result<(), ()> {
     assert_eq!(val0, val1);
     Ok(())
 }
+
+#[test]
+fn vec_front_and_back() -> Result<(), ()> {
+    let mut host = Host::default();
+    let scvec0: ScVec = ScVec(vec![ScVal::U32(1), ScVal::U32(2), ScVal::U32(3)].try_into()?);
+    let scobj0 = ScObject::Vec(scvec0);
+    let obj0 = host.to_host_obj(&scobj0).unwrap();
+    let front = unsafe {
+        <i32 as RawValConvertible>::unchecked_from_val(host.vec_front(*obj0.as_ref()).unwrap())
+    };
+    let back = unsafe {
+        <i32 as RawValConvertible>::unchecked_from_val(host.vec_back(*obj0.as_ref()).unwrap())
+    };
+    assert_eq!(front, 1);
+    assert_eq!(back, 3);
+    Ok(())
+}
+
+#[test]
+#[should_panic(expected = "value does not exist")]
+fn empty_vec_front_and_back() {
+    let mut host = Host::default();
+    let scvec0: ScVec = ScVec(vec![].try_into().unwrap());
+    let scobj0 = ScObject::Vec(scvec0);
+    let obj0 = host.to_host_obj(&scobj0).unwrap();
+    host.vec_front(*obj0.as_ref()).unwrap();
+    host.vec_back(*obj0.as_ref()).unwrap();
```

Comment:
Only the front call is being tested in this test because the test is failing immediately. You'll either need to make two tests, or use catch_unwind to catch the exception and continue.

Reply:
Good point! 
`catch_unwind` doesn't work complaining some types not being unwind safe. Even if it does, in order to compare error message (not just checking if `is_err()`), we would need to add `PartialEq` trait in a few places including in the XDR.
Anyway I think the better approach is to test each case separately. See update.
I've added `rstest` and used its `fixture` to avoid code repetition.